### PR TITLE
Updates to the CHANGELOG from the last 3 weeks.

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

**Review by breedx-splk — APPROVED** [review]
Looks good to me.  Would be rad to link to the PRs where relevant.  🤷🏻

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,21 @@
 
 ## Unreleased:
 
+### API
+#### Enhancements
+- Parsing of the W3C Baggage header has been optimized.
+
+### SDK
```
Comment (anuraaga) [inline CHANGELOG.md:9]: Think we need to callout the small behavior change in https://github.com/open-telemetry/opentelemetry-java/pull/3244

**Review by anuraaga — APPROVED** [review]
My Update Dependencies PR worked? :)

**Comment by jkwatson** [comment]
> Looks good to me. Would be rad to link to the PRs where relevant. 🤷🏻

sounds like someone wants to help maintain the CHANGELOG!

File: CHANGELOG.md (hunk 1 above)
Comment (jkwatson) [reply CHANGELOG.md:9]: i'll add it. good call.

**Review by breedx-splk — APPROVED** [review]

**Pull request merged**